includes/Modules/Analytics.php

```diff
@@ -905,44 +930,68 @@ function ( $date_range ) {
 				);
 				$request->setDateRanges( $date_ranges );
 
-				$metrics = array_map(
-					function ( $metric_def ) {
-						$metric_def = array_merge(
-							array(
-								'alias'      => '',
-								'expression' => '',
-							),
-							(array) $metric_def
-						);
-						$metric     = new Google_Service_AnalyticsReporting_Metric();
-						$metric->setAlias( $metric_def['alias'] );
-						$metric->setExpression( $metric_def['expression'] );
+				$metrics = $data['metrics'];
+				if ( ! empty( $metrics ) && ( is_string( $metrics ) || is_array( $metrics ) ) ) {
+					if ( is_string( $metrics ) ) {
+						$metrics = explode( ',', $data['metrics'] );
+					} elseif ( is_array( $metrics ) && ! wp_is_numeric_array( $metrics ) ) { // If single object is passed.
+						$metrics = array( $metrics );
+					}
 
-						return $metric;
-					},
-					(array) $data['metrics']
-				);
-				$request->setMetrics( $metrics );
+					$metrics = array_filter(
+						array_map(
+							function ( $metric_def ) {
+								$metric = new Google_Service_AnalyticsReporting_Metric();
+
+								if ( is_string( $metric_def ) ) {
+									$metric->setAlias( $metric_def );
+									$metric->setExpression( $metric_def );
+								} elseif ( is_array( $metric_def ) && ! empty( $metric_def['expression'] ) ) {
+									$metric->setExpression( $metric_def['expression'] );
+									$metric->setAlias( ! empty( $metric_def['alias'] ) ? $metric_def['alias'] : $metric_def['expression'] );
+								} else {
+									return null;
+								}
+
+								return $metric;
+							},
+							array_filter( $metrics )
+						)
+					);
+
+					if ( ! empty( $metrics ) ) {
+						$request->setMetrics( $metrics );
+					}
+				}
 
 				// Order by.
-				$orderby = array_map(
-					function ( $order_def ) {
-						$order_def = array_merge(
-							array(
-								'fieldName' => '',
-								'sortOrder' => '',
-							),
-							(array) $order_def
-						);
-						$order_by  = new Google_Service_AnalyticsReporting_OrderBy();
-						$order_by->setFieldName( $order_def['fieldName'] );
-						$order_by->setSortOrder( $order_def['sortOrder'] );
+				$orderby = $data['orderby'];
+				if ( ! empty( $orderby ) && is_array( $orderby ) ) {
+					// When just object is passed we need to convert it to an array of objects.
+					if ( ! is_array( $orderby[0] ) ) {
+						$orderby = array( $orderby );
+					}
 
-						return $order_by;
-					},
-					(array) $data['orderby']
-				);
-				$request->setOrderBys( $orderby );
+					$orderby = array_map(
+						function ( $order_def ) {
+							$order_def = array_merge(
+								array(
+									'fieldName' => '',
+									'sortOrder' => '',
```

Comment:
Maybe we should specify "DESCENDING" here as default, to ensure this is never empty for a request?

Reply:
Updated.